Kudu.Core/Settings/DiagnosticsSettingsManager.cs

```diff
@@ -87,6 +93,7 @@ private void SaveSettings(DiagnosticsSettings diagnosticsSettings)
                 FileSystemHelpers.EnsureDirectory(Path.GetDirectoryName(_path));
             }
 
+            diagnosticsSettings.EnsureValid();
```

Comment:
Why does this have to be done after deserialization? Won't the TraceEnum converter validate these settings for us on deserialization? If not, we could have our own converter that does. That way, when we add new settings we don't have to remember to add the setting key name to your static list.